Updates for version files

KateFriedman-NOAA opened the pull request:
Several updates related to new versions files:
* Add build.ver and run.ver to new versions subfolder
* Adjust path to run.ver in downstream scripts (did not test change)
* Change module/library versions in module_base.wcoss2 to variables that come from above
* A few updates to run.ver based on code manager feedback
* Update $FIX_DIR to newer temporary global location (not enough space in EIB space to hold all needed fix sets)
* Update to new GLDAS tag that includes version variable overrides

KateFriedman-NOAA committed:
Add WCOSS2 to load_fv3gfs_modules.sh

- add WCOSS2 stanza to machine check if-block in load_fv3gfs_modules.sh
- add load of module_base.wcoss2

Refs: #399

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:feature/ops-wcoss2' into feature/ops-wcoss2

KateFriedman-NOAA committed:
Rename and move run.version file to run.ver

- rename "run.version" to "run.ver"
- move file to new "versions" folder at top level

Refs: #399

KateFriedman-NOAA committed:
Update versions file path in downstream drivers

- run.version was renamed to run.ver and moved up to a new
versions folder at the top level
- update location of run versions file in downstream driver scripts

Refs: #399

KateFriedman-NOAA committed:
Adjust run.ver path to be relative to $driver

- for the driver scripts use $driver variable to set path to run.ver

Refs: #399

KateFriedman-NOAA committed:
New versions file updates

- add new build.ver file to set module/library versions at buildtime
- update run.ver file to reorganize versions and add missing ones needed
at runtime
- update machine-setup.sh to source versions/build.ver at buildtime
- update module_base.wcoss2 to accept module/library version overrides
from the run.ver file at runtime

Refs: #399

KateFriedman-NOAA committed:
Add COMPATH path to support WAFS

- Yali Mao requsted that the following path be added to COMPATH to
support the WAFS: /lfs/h1/ops/canned/com/hourly
- added to run.ver

Refs: #399

KateFriedman-NOAA committed:
Update GLDAS tag to gldas_gfsv16_release.v1.19.0

Refs: #399

KateFriedman-NOAA committed:
Link script updates for fix and global_chgres

- update FIX_DIR on WCOSS2 for new global location (still temporary)
- remove global_chgres from the ufs_utilsexe exec list; the exec no
longer comes from UFS_UTILS so no longer need the symlink for it

Refs: #399

KateFriedman-NOAA committed:
Load run version file updates

- add source of run.ver to module load script
- source module_base.wcoss2 instead of loading
- change USE_CFP setting in module_base.wcoss2 from setenv to export

Refs: #399

BoiVuong-NOAA reviewed (approved):
look good

KateFriedman-NOAA committed:
Revert wgrib2 to v2.0.7 in version files

- the UPP needs to use wgrib2/2.0.7 instead of wgrib2/2.0.8
- decision to use wgrib2/2.0.7 in WCOSS2 port of GFSv16 was made between
UPP and NCO managers
- update both build.ver and run.ver to set wgrib2/2.0.7

Refs; #399

KateFriedman-NOAA committed:
Convert module_base.wcoss2 to LUA

- rename module_base.wcoss2 to module_base.wcoss2.lua
- convert contents of module_base.wcoss2.lua to LUA format
- update load_fv3gfs_modules.sh to load modulefile again instead of
sourcing it like a bash script
- add set +x and set -x to module load section of load_fv3gfs_modules.sh

Refs: #399

KateFriedman-NOAA commented:
@aerorahul @WalterKolczynski-NOAA I further updated this PR to include updates to convert module_base.wcoss2 into LUA format (now named module_base.wcoss2.lua). Please review this PR and let me know how the LUA changes look. I tested the new module_base.wcoss2.lua in a gfspost job on Cactus and it correctly loaded the runtime modules. Thanks!

@aerorahul The component code managers still need to be poked about converting to LUA as well, thanks!

WalterKolczynski-NOAA commented on modulefiles/module_base.wcoss2.lua:
```diff
@@ -0,0 +1,62 @@
+Load environment to run GFS on WCOSS2
+]])
+
+PrgEnv_intel_ver=os.getenv("PrgEnv_intel_ver")
```
Since these are each only being used once, do we want to just use os.getenv in-line in the load statements instead of loading them to variables first?

KateFriedman-NOAA replied:
Addressed.

WalterKolczynski-NOAA commented:
When this comes back to develop, we should update all the modules to be lua.

KateFriedman-NOAA committed:
Compress contents of new module_base.wcoss2.lua

- rework new module_base.wcoss2.lua to compress the version variable grab
within the load line

Refs: #399

KateFriedman-NOAA commented:
> When this comes back to develop, we should update all the modules to be lua.

Definitely! I plan to take lessons learned and results of the changes for the ops side into the same updates for develop. Haven't started that effort yet.

KateFriedman-NOAA commented:
Thanks @WalterKolczynski-NOAA !
